python/ray/scripts/scripts.py

```diff
@@ -409,22 +409,34 @@ def stop():
     # Note that raylet needs to exit before object store, otherwise
     # it cannot exit gracefully.
     processes_to_kill = [
-        "raylet",
-        "plasma_store_server",
-        "raylet_monitor",
-        "monitor.py",
-        "redis-server",
-        "default_worker.py",  # Python worker.
-        " ray_",  # Python worker.
-        "org.ray.runtime.runner.worker.DefaultWorker",  # Java worker.
-        "log_monitor.py",
-        "reporter.py",
-        "dashboard.py",
+        # The first element is to filter ps results by command name (only the
+        # executable name). The second element is to filter ps results by
+        # command with all its arguments. See STANDARD FORMAT SPECIFIERS
+        # section of http://man7.org/linux/man-pages/man1/ps.1.html about comm
+        # and args. The two levels of filters can help avoid killing non-ray
+        # processes.
```

Comment:
Do you think it'd be clearer to put them in 2 separate lists? e.g., `processes_to_kill_by_comm` and `processes_to_kill_by_args`

Reply:
The order matters. If you think [XXX, None] and [None, YYY] is not easy to understand, I can change it to [XXX, 'comm'] and [YYY, 'args'].

Reply:
I see. What about `[XXX, True]` and `[YYY, False]`?

Reply:
Sure. Done.